src/System.Management.Automation/engine/hostifaces/PowerShellProcessInstance.cs

```diff
@@ -58,7 +58,7 @@ public PowerShellProcessInstance(Version powerShellVersion, PSCredential credent
                     CultureInfo.InvariantCulture,
                     "{0} -wd \"{1}\"",
                     processArguments,
-                    workingDirectory.Replace("\"", "\"\""));
+                    workingDirectory.Replace("\"", "\"\"").Replace("\\", "\\\\"));
```

Comment:
No side effects for native command running?
/cc @mklement0

Reply:
I don't know what you mean by a 'native running command'.  The provided working directory path is only for session drive location.  I am not aware of any issues with escaping back slashes, but would like to know if you have a specific concern.

Reply:
I don't even see the problem, at least from _within_ PowerShell - what is the exact failure scenario?

The following test works fine for me in PowerShell Core 7.0.0-preview.5 on Windows:

```powershell
Start-Job { $pwd } -WorkingDirectory '\'  | Receive-Job -Wait -AutoRemoveJob | Should -Be ('/', 'c:\')[$IsWindows]
```

It does fail on Unix, but due to an unrelated bug: the `-WorkingDirectory` parameter only supports _native_ paths - see #11050

Reply:
This fails on the latest build of preview.5, on Windows only.  Windows command line parsing removes odd numbered trailing back slash.

```powershell
PS C:\> Start-Job { $pwd } -WorkingDirectory '\'  | Receive-Job -Wait -AutoRemoveJob
Set-Location: Cannot find path '"' because it does not exist.
OpenError: Running startup script threw an error: Cannot find path '"' because it does not exist..
```

Reply:
I have no understanding the fix or the PR description :-(
From the PR description:
> Windows command line parsing will drop any trailing back slash in the argument list. 

Why do this cause the problem? Set-Location works well with and without trailing slashes, and with escaped slashes too. Maybe command line parser eats all back slashes?

Reply:
I see - this happened _post_-preview.5, due to #10951 (worth mentioning in this PR's description).

The more robust fix would actually be to switch to using `ProcessStartInfo.ArgumentList` instead of `ProcessStartInfo.Arguments` and let CoreFx do the proper escaping.

Reply:
@iSazonov: The issue is that a string with _verbatim_ content such as `"\"` breaks Windows command-line parsing, because the `\"` is (rightfully) interpreted as an _escaped_ `"` char. - this is the same problem that led to the `-RemoveWorkingDirectoryTrailingCharacter` workaround in the CLI - see https://github.com/PowerShell/PowerShell/pull/8287

Reply:
@PaulHigin - Can you respond to @mklement0 comment about switching to `ProcessStartInfo.ArgumentList` instead of `ProcessStartInfo.Arguments`

Reply:
The quote is not necessary for the repro.  Command line document states that odd numbered trailing back slash is removed, which is the case here.  I thought that escaping all back slashes would be the right fix, but let me know if there is a problem with that.

I feel that this fix should address the bug, and changing to use ArgumentList is out of scope.  Please create a separate Issue for that.

Reply:
@PaulHigin: `ProcessStartInfo.ArgumentList` is definitely the way to go, but fair point re creating a new issue. 

> The quote is not necessary for the repro. 

It _is_ - that's why the problem didn't exist pre-#10951 - a lone `\` at the end of an _unquoted_ argument is _not_ a problem.

For instance, `foo.exe a\ b` causes foo.exe to properly see arguments `a\` and `b`.
By contrast, `foo.exe "a\" b` breaks (foo.exe sees single argument `a" b`).

Reply:
Re switch to `ProcessStartInfo.ArgumentList`: Since another fix is pending anyway, I've added a note to #11050

Reply:
Final thought (emphasis added):

> I thought that escaping all back slashes would be the _right_ fix

It is an _effective_ fix, but note that the right fix would have been to only double the _trailing_ backslash.
The fix is still effective, because Windows happily accepts (verbatim) paths such as `c:\\temp\` (arbitrary duplication of the path separator).

These are the kinds of subtleties that `ProcessStartInfo.ArgumentList` handles automatically.

Reply:
Ok, you've convinced me :).  I'll update the fix to use ArgumentList.